bokeh/tests/selenium/config.py

```diff
@@ -0,0 +1,215 @@
+                print("\tversion:  %s" % (cap.version))
+                if not _counter == 0:
+                    print("")
+                _counter += 1
+
+    def get_default_browser_caps(self, browser):
+        return self.distributive_browsers_caps[browser][0]
+
+    def get_default_browser_cap(self, browser, capability='platform'):
+        if capability == 'platform':
+            return self.distributive_browsers_caps[browser][0].platform
+        elif capability == 'version':
+            return self.distributive_browsers_caps[browser][0].version
+        else:
+            print("Internal error, no support for capability: %s" % (cap_name))
+            sys.exit(1)
+
+
+class TestSettings(object):
+
+    def __init__(self, metadata=None, **kwargs):
+        if not metadata:
+            self.metadata = TestMetadata()
+        else:
+            self.metadata = metadata
+
+        if 'env_mode' in kwargs:
+            self.env_mode = kwargs.get('env_mode')
+        else:
+            self.env_mode = None
+
+        if not self.env_mode:
+            self.env_mode = 'standalone'
+
+        if not check_for_proper_arg(self.env_mode, self.metadata.env_modes):
+            print("Invalid testing environment type: %s" % (self.env_mode))
+            sys.exit(1)
+
+        if 'browser_engine' in kwargs:
+            self.browser_engine = kwargs.get('browser_engine')
+        else:
+            self.browser_engine = None
+
+        if not self.browser_engine:
+            self.browser_engine = self.metadata.browser_engine
+
+        if not check_for_proper_arg(self.browser_engine, self.metadata.browser_engines):
+            print("Invalid browser_engine: %s" % (self.browser_engine))
+            sys.exit(1)
+
+        if 'browser_caps' in kwargs:
+            self.browser_caps = kwargs.get('browser_caps')
+        else:
+            self.browser_caps = None
+
+        if not self.browser_caps:
+            self.browser_caps = self.metadata.browser_caps
+
+        self.standalone_browsers_bin = self.metadata.standalone_browsers_bin.copy()
+
+        if 'standalone_browsers_bin' in kwargs:
+            for key, value in iter(kwargs.get('standalone_browsers_bin').items()):
+
+                if key in self.standalone_browsers_bin:
+                    self.standalone_browsers_bin.update({key:value})
+                else:
+                    print("Non existent 'browser_engine' key: %s" % (key))
+                    sys.exit(1)
+
+        if 'selenium_server_jar_path' in kwargs:
+            self.selenium_server_jar_path = kwargs.get('selenium_server_jar_path')
+        else:
+            self.selenium_server_jar_path = None
+
+        if not self.selenium_server_jar_path:
+            self.selenium_server_jar_path = self.metadata.selenium_server_jar_path
+
+        if 'data_dir' in kwargs:
+            self.data_dir = kwargs.get('data_dir')
+        else:
+            self.data_dir = None
```

Comment:
All of these `if...else` blocks should be expressible as just `kwargs.get('data_dir', None)`